### ISSUE TITLE
MA0050 is not reported when method using new static throw methods on ArgumentException

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#616]
### Version of the Meziantou.Analyzer NuGet package

2.0.93

### Rule Identifier

MA0050

### Target Framework

net7.0

### C# Language version

C# 11

### Description

When a iterator method uses the "new" static throw methods on **ArgumentException** (and derived classes) to validate arguments, then **MA0050** is not reported.

### Reproduction Steps

**Minimal code**

```cs
using System.Runtime.CompilerServices;
using System.Threading;

public static class Program
{
    private static async Task Main()
    {
        var cancellationTokenSource = new CancellationTokenSource();
        await cancellationTokenSource.CancelAsync().ConfigureAwait(false);

        await foreach (string x in Run("x").WithCancellation(cancellationTokenSource.Token).ConfigureAwait(false))
        {
            Console.WriteLine(x);
        }
    }

    private static async IAsyncEnumerable<string> Run(string? value, [EnumeratorCancellation] CancellationToken cancellationToken = default)
    {
        ArgumentNullException.ThrowIfNull(value);
        ArgumentException.ThrowIfNullOrWhiteSpace(value);
        ArgumentOutOfRangeException.ThrowIfNullOrEmpty(value);

        await Task.Delay(1000, cancellationToken).ConfigureAwait(false);

        yield return "X";
        yield return "Y";
    }
}
```

**Expected result**

MA0050 is reported.

**Actual result**

MA0050 is not reported.

**Comment by meziantou** [comment]
Fixed by 097108358aaf899f4430b4353ffdd9bd7db100e2
